stackslib/src/chainstate/burn/db/sortdb.rs

```diff
@@ -2909,6 +2909,12 @@ impl SortitionDB {
         SortitionDB::apply_schema_6(&db_tx, epochs_ref)?;
         SortitionDB::apply_schema_7(&db_tx, epochs_ref)?;
         SortitionDB::apply_schema_8_tables(&db_tx, epochs_ref)?;
+        // `apply_schema_8_migration` creates new transactions, so
+        // commit this first.
+        db_tx.commit()?;
+        // NOTE: we don't need to provide a migrator here because we're not migrating
+        self.apply_schema_8_migration(None)?;
```

Comment:
Ah, this might be what broke everything.  You need to move `apply_schema_8_migration(None)` back to where it was, _after_ `db_tx.instantiate_index()`.  Then, after that, you can run `apply_schema_9`.

Reply:
Yep, that was it! Also needed `insert_block_snapshot`. The few tests I tried passed locally, so hopefully things look good after this.